hrms/payroll/doctype/salary_slip/salary_slip.py

```diff
@@ -1173,19 +1188,21 @@ def update_component_row(
 
 		component_row.amount = amount
 
-		self.update_component_amount_based_on_payment_days(component_row)
+		self.update_component_amount_based_on_payment_days(component_row, allow_zero_value_component)
 
 		if data:
 			data[component_row.abbr] = component_row.amount
 
-	def update_component_amount_based_on_payment_days(self, component_row):
+	def update_component_amount_based_on_payment_days(
+		self, component_row, allow_zero_value_component=None
+	):
 		joining_date, relieving_date = self.get_joining_and_relieving_dates()
 		component_row.amount = self.get_amount_based_on_payment_days(
 			component_row, joining_date, relieving_date
 		)[0]
 
 		# remove 0 valued components that have been updated later
-		if component_row.amount == 0:
+		if component_row.amount == 0 and not allow_zero_value_component:
```

Comment:
There was one case for which this explicit removal was added. Check if this case still works fine

https://github.com/frappe/erpnext/pull/31425

Add a test case if possible

Reply:
If the setting is enabled then the system will include zero-valued components. maybe instead of a global setting, we can allow component-based configuration